ECommerceAPI/Repositories/Implementations/ProductRepository.cs

```diff
@@ -60,5 +63,58 @@ public async Task<bool> DeleteProductAsync(long productId)
             await context.SaveChangesAsync();
             return true;
         }
+        public async Task<AttributeValue?> GetAttributeValueByIdAsync(long valueId)
+        {
+            var value = await context.AttributeValues
+                .AsNoTracking()
+                .Where(v => v.Id == valueId)
+                .FirstOrDefaultAsync();
+            return value;
+        }
+        public async Task<bool> AddAttributeToProductAsync(ProductAttributeValue attributeValue)
+        {
+            context.ProductAttributeValues.Add(attributeValue);
+            await context.SaveChangesAsync();
+            return true;
+        }
+        public async Task<IList<ProductAttributeValue>> GetProductAttributesAsync(long productId)
+        {
+            var attributes = await context.ProductAttributeValues
+                .Include(pav => pav.Product)
+                .Include(pav => pav.AttributeValue)
+                .AsNoTracking()
+                .Where(pav => pav.ProductId == productId)
+                .ToListAsync();
+            return attributes;
+        }
+        public async Task<bool> UpdateProductAttributeAsync(ProductAttributeValue productAttributeValue)
+        {
+            var attribute = await context.ProductAttributeValues
+                .Where(pav => pav.ProductId == productAttributeValue.ProductId)
+                .Where(pav => pav.AttributeValueId == productAttributeValue.AttributeValueId)
+                .FirstOrDefaultAsync();
+            if (attribute is null)
+            {
+                return false;
+            }
+            attribute.Quantity = productAttributeValue.Quantity;
+            context.ProductAttributeValues.Update(attribute);
+            await context.SaveChangesAsync();
+            return true;
+        }
+        public async Task<bool> DeleteProductAttributeAsync(long productId, long attributeValueId)
+        {
+            var attribute = await context.ProductAttributeValues
+                .Where(pav => pav.ProductId == productId)
+                .Where(pav => pav.AttributeValueId == attributeValueId)
+                .FirstOrDefaultAsync();
+            if (attribute is null)
```

Comment:
double Where is not good, .Where(pav => pav.ProductId == productId && pav.AttributeValueId == attributeValueId )